src/node.rs

```diff
@@ -616,51 +622,93 @@ impl PbftNode {
                     .get_unvalidated_block_with_id(block.previous_id.as_slice())
             });
         if previous_block.is_none() {
-            self.service
-                .fail_block(block.block_id.clone())
-                .unwrap_or_else(|err| error!("Couldn't fail block due to error: {:?}", err));
-            return Err(PbftError::InternalError(format!(
-                "Received block {:?} / {:?} but node does not have previous block {:?}",
-                block.block_num,
-                hex::encode(&block.block_id),
-                hex::encode(&block.previous_id),
-            )));
-        }
+            //It means that the block has come out of sequence .  We should not fail the  block,
+            //rather we will push it into buffer so that we can use it when it's previous block arrives.
 
-        // Make sure that the previous block has the previous block number (enforces that blocks
-        // are strictly monotically increasing by 1)
-        let previous_block = previous_block.expect("Previous block's existence already checked");
-        if previous_block.block_num != block.block_num - 1 {
-            self.service
-                .fail_block(block.block_id.clone())
-                .unwrap_or_else(|err| error!("Couldn't fail block due to error: {:?}", err));
-            return Err(PbftError::InternalError(format!(
-                "Received block {:?} / {:?} but its previous block ({:?} / {:?}) does not have \
-                 the previous block_num",
+            info!(
+                "{}: Got BlockNew out of sequence: {} / {}",
+                state,
                 block.block_num,
-                hex::encode(&block.block_id),
-                block.block_num - 1,
-                hex::encode(&block.previous_id),
-            )));
-        }
+                hex::encode(&block.block_id)
+            );
+            trace!("checking consensus seal by verifying that the block has a payload field");
```

Comment:
This debug statement does not seem to provide any useful information.

Reply:
Noted. Thanks.